From 3339013b72d76906195a347fc2f71f48987b143a Mon Sep 17 00:00:00 2001
From: Krutika Dhananjay <kdhananj@redhat.com>
Date: Mon, 9 May 2016 16:53:19 +0530
Subject: [PATCH 185/192] storage/posix: Print offset,size and gfid too when readv fails

... for better debuggability.

        Backport of: http://review.gluster.org/14271
        3.7 backport: http://review.gluster.org/14625

Here's a sample log:

[2016-05-31 05:18:34.893951] E [MSGID: 113040] [posix.c:3089:posix_readv]
0-dis-posix: read failed on gfid=3744d34b-f798-4ce2-b421-cc632a9903bd,
fd=0x7f8b3c015abc, offset=131072 size=131072 [Invalid argument]

Change-Id: I35040a6c1e5108afd44ab49f4406fcd291465b64
BUG: 1339136
Signed-off-by: Krutika Dhananjay <kdhananj@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/75846
Reviewed-by: Pranith Kumar Karampuri <pkarampu@redhat.com>
Tested-by: Pranith Kumar Karampuri <pkarampu@redhat.com>
---
 xlators/storage/posix/src/posix.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/xlators/storage/posix/src/posix.c b/xlators/storage/posix/src/posix.c
index d5c18f1..e19da1a 100644
--- a/xlators/storage/posix/src/posix.c
+++ b/xlators/storage/posix/src/posix.c
@@ -3027,7 +3027,9 @@ posix_readv (call_frame_t *frame, xlator_t *this,
         if (op_ret == -1) {
                 op_errno = errno;
                 gf_msg (this->name, GF_LOG_ERROR, errno, P_MSG_READ_FAILED,
-                        "read failed on fd=%p", fd);
+                        "read failed on gfid=%s, fd=%p, offset=%"PRIu64" "
+                        "size=%"GF_PRI_SIZET"", uuid_utoa (fd->inode->gfid), fd,
+                        offset, size);
                 goto out;
         }
 
-- 
1.7.1

